homeassistant/components/hassio/update.py

```diff
@@ -0,0 +1,288 @@
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import (
+    ADDONS_COORDINATOR,
+    async_update_addon,
+    async_update_core,
+    async_update_os,
+    async_update_supervisor,
+)
+from .const import (
+    ATTR_CHANGELOG,
+    ATTR_VERSION,
+    ATTR_VERSION_LATEST,
+    DATA_KEY_ADDONS,
+    DATA_KEY_OS,
+    DATA_KEY_SUPERVISOR,
+)
+from .entity import (
+    HassioAddonEntity,
+    HassioCoreEntity,
+    HassioOSEntity,
+    HassioSupervisorEntity,
+)
+from .handler import HassioAPIError
+
+ENTITY_DESCRIPTION = UpdateEntityDescription(
+    name="Update",
+    key=ATTR_VERSION_LATEST,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Supervisor update based on a config entry."""
+    coordinator = hass.data[ADDONS_COORDINATOR]
+
+    entities = [
+        SupervisorSupervisorUpdateEntity(
+            coordinator=coordinator,
+            entity_description=ENTITY_DESCRIPTION,
+        ),
+        SupervisorCoreUpdateEntity(
+            coordinator=coordinator,
+            entity_description=ENTITY_DESCRIPTION,
+        ),
+    ]
+
+    for addon in coordinator.data[DATA_KEY_ADDONS].values():
+        entities.append(
+            SupervisorAddonUpdateEntity(
+                addon=addon,
+                coordinator=coordinator,
+                entity_description=ENTITY_DESCRIPTION,
+            )
+        )
+
+    if coordinator.is_hass_os:
+        entities.append(
+            SupervisorOSUpdateEntity(
+                coordinator=coordinator,
+                entity_description=ENTITY_DESCRIPTION,
+            )
+        )
+
+    async_add_entities(entities)
+
+
+class SupervisorAddonUpdateEntity(HassioAddonEntity, UpdateEntity):
+    """Update entity to handle updates for the Supervisor add-ons."""
+
+    _attr_supported_features = UpdateEntityFeature.INSTALL | UpdateEntityFeature.BACKUP
+
+    @property
+    def title(self) -> str | None:
+        """Return the title of the update."""
+        if not self.available:
+            return None
+        return self.coordinator.data[DATA_KEY_ADDONS][self._addon_slug][ATTR_NAME]
+
+    @property
+    def latest_version(self) -> str | None:
+        """Latest version available for install."""
+        if not self.available:
+            return None
+        return self.coordinator.data[DATA_KEY_ADDONS][self._addon_slug][
+            ATTR_VERSION_LATEST
+        ]
+
+    @property
+    def current_version(self) -> str | None:
+        """Version currently in use."""
+        if not self.available:
+            return None
+        return self.coordinator.data[DATA_KEY_ADDONS][self._addon_slug][ATTR_VERSION]
+
+    @property
+    def release_summary(self) -> str | None:
+        """Release summary for the add-on."""
+        if self.current_version == self.latest_version:
+            return None
```

Comment:
Is this needed? As in, showing the last release notes isn't bad?

```suggestion
```

Reply:
Small attempt to keep the DB writes lower, but no, not needed.